File: app/src/test/java/dictionary/DictionaryTest.java
package dictionary;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.io.File;
import java.io.FileNotFoundException;
import java.net.URL;
import java.nio.file.Path;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.io.TempDir;

import dictionary.Dictionary.DictionaryBuilder;

public class DictionaryTest {

    @Test
    public void createCompleteDictionary(@TempDir Path tempDir) throws FileNotFoundException {
        URL words = getClass().getClassLoader().getResource("definitions.txt");
        DictionaryBuilder builder = new DictionaryBuilder(tempDir.toString() + File.separator);
        assertDoesNotThrow(() -> {
            builder.cover("title", "createdBy")
                    .copyright()
                    .content(words.getPath())
                    .build();
        });

        File[] outputFiles = tempDir.toFile().listFiles();
        assertEquals(3, outputFiles.length);
        List<String> fileNames = Stream.of(outputFiles).map(File::getName).collect(Collectors.toList());
        assertTrue(fileNames.contains(CoverPage.FILE_NAME));
        assertTrue(fileNames.contains(CopyrightPage.FILE_NAME));
        assertTrue(fileNames.contains(ContentPage.FILE_NAME));
    }

    @Test
    public void noFiles(@TempDir Path tempDir){
        assertDoesNotThrow(() -> new DictionaryBuilder(tempDir.toString() + File.separator).build());
        File[] outputFiles = tempDir.toFile().listFiles();
        assertEquals(0, outputFiles.length);
    }

}

File: app/src/test/java/dictionary/DictionaryTest.kt
package dictionary

import dictionary.Dictionary.DictionaryBuilder
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.io.TempDir
import java.io.File
import java.io.FileNotFoundException
import java.nio.file.Path
import java.util.stream.Collectors
import java.util.stream.Stream

class DictionaryTest {
    @Test
    @Throws(FileNotFoundException::class)
    fun createCompleteDictionary(@TempDir tempDir: Path) {
        val words = this::class.java.classLoader.getResource("definitions.txt")
        val builder = DictionaryBuilder(tempDir.toString() + File.separator)
        Assertions.assertDoesNotThrow {
            builder.cover("title", "createdBy")
                .copyright()
                .content(words!!.path)
                .build()
        }
        val outputFiles = tempDir.toFile().listFiles()
        Assertions.assertEquals(3, outputFiles?.size)
        val fileNames = Stream.of(*outputFiles).map { obj: File -> obj.name }
            .collect(Collectors.toList())
        Assertions.assertTrue(fileNames.contains(CoverPage.FILE_NAME))
        Assertions.assertTrue(fileNames.contains(CopyrightPage.FILE_NAME))
        Assertions.assertTrue(fileNames.contains(ContentPage.FILE_NAME))
    }

    @Test
    fun noFiles(@TempDir tempDir: Path) {
        Assertions.assertDoesNotThrow<Dictionary> { DictionaryBuilder(tempDir.toString() + File.separator).build() }
        val outputFiles = tempDir.toFile().listFiles()
        Assertions.assertEquals(0, outputFiles?.size)
    }
}